contracts/halo-pool/src/state.rs

```diff
@@ -37,51 +29,20 @@ pub struct RewardTokenAsset {
     pub amount: Uint128,
 }
 
+#[cw_serde]
+pub struct RewardTokenAssetResponse {
+    pub info: TokenInfo,
+    pub amount: Uint128,
+    pub time_query: u64,
+}
+
 impl fmt::Display for RewardTokenAsset {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
         write!(f, "{}: {}", self.info, self.amount)
     }
 }
 
 impl RewardTokenAsset {
-    pub fn new(info: TokenInfo, amount: Uint128) -> Self {
-        Self { info, amount }
-    }
-
-    pub fn is_token(&self) -> bool {
-        self.info.is_token()
-    }
-
-    pub fn is_native_token(&self) -> bool {
-        self.info.is_native_token()
-    }
-
-    pub fn into_msg(self, recipient: Addr) -> StdResult<CosmosMsg> {
-        let amount = self.amount;
-
-        match &self.info {
-            TokenInfo::Token { contract_addr } => Ok(CosmosMsg::Wasm(WasmMsg::Execute {
-                contract_addr: contract_addr.to_string(),
-                msg: to_binary(&Cw20ExecuteMsg::Transfer {
-                    recipient: recipient.to_string(),
-                    amount,
-                })?,
-                funds: vec![],
-            })),
-            TokenInfo::NativeToken { denom } => Ok(CosmosMsg::Bank(BankMsg::Send {
-                to_address: recipient.to_string(),
-                amount: vec![Coin {
-                    amount: self.amount,
-                    denom: denom.to_string(),
-                }],
-            })),
-        }
-    }
-
-    pub fn into_submsg(self, recipient: Addr) -> StdResult<SubMsg> {
-        Ok(SubMsg::new(self.into_msg(recipient)?))
-    }
-
     pub fn assert_sent_native_token_balance(&self, message_info: &MessageInfo) -> StdResult<()> {
         if let TokenInfo::NativeToken { denom } = &self.info {
             match message_info.funds.iter().find(|x| x.denom == *denom) {
```

Comment:
Chỗ này dùng cosmwasm_std::has_coins cho gọn

Reply:
Updated to use has_coins